### ISSUE TITLE
Integrate with AWS Distro for OpenTelemetry

### ISSUE THREAD

**Issue by eduardomourar** [awslabs/aws-lambda-powertools-roadmap#1]
* Why [AWS Distro for OpenTelemetry](https://aws.amazon.com/otel/)?

> OpenTelemetry provides open source APIs, libraries, and agents to instrument, generate, collect, and export telemetry data (metrics, logs, and traces) of distributed applications.

Because there is a few overlap between the lambda power tools and OpenTelemetry, it would be nice to see how the layer provided can be reused or merged:
https://github.com/open-telemetry/opentelemetry-lambda/tree/main/python

Another important aspect, although it has not been fully finalized yet, is to follow the [Log Data Model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md) defined in the OpenTelemetry spec so that the context can be later related to both metric and trace.

**Comment by heitorlessa** [comment]
Hey @eduardomourar thanks a lot for the suggestion. This is something we'd like to consider for v2. I signalled this to the OpenTelemetry team to have a discussion on how to integrate in the future - I'll leave as `Ideas` until we have an initial chat with the OTel team on this in H2.

Following Log Data Model until it's finalized, stable, and in use for some time would be a good measure too besides customers +1 this feature request. In the medium term (v2, Q4), I can see us providing a better integration but not using it as the default yet.

**Comment by go4real** [comment]
It would be nice if we could change the logging backend between the open telemetry agent and the x-ray agent by setting some configuration values without any code changes.

There are currently two options for tracking data, each with pros and cons. And our main concern that is hesitating to do more custom instrumentation is code changes between logging backend changes. If there is a wrapper sdk, we will use it without hesitation.

**Comment by gbisaga** [comment]
I agree, this would be great. We want to be committed to Otel in the long run, but the native integration with lambdas is a winner. Having a switchable backend would let large organizations commit to this library knowing it could be used inline with longer term plans.

**Comment by heitorlessa** [comment]
Sharing updates as we're discussing the Observability Provider feature in Python: https://github.com/awslabs/aws-lambda-powertools-python/issues/1433

As of now, we no longer intend to move entirely to OTel (ADOT included) for two main reasons:

* Cold start increase is too significant because it requires a Daemon to be spun up (Java, .NET being the most impacted)
* Logging is still in [Draft](https://opentelemetry.io/docs/reference/specification/status/#current-status) status

Instead, we will focus on making it easier to bring any observability SDK/agent through observability providers (e.g., we have an undocumented `TracingProvider` today).

**Comment by willfarrell** [comment]
@Dilski Take a read of a similar thread with a more recent update on this: https://github.com/awslabs/aws-lambda-powertools-typescript/discussions/665#discussioncomment-3834638

> ... we won't be making any plans for moving to OTel at least in 2023.
We'll communicate with ample time if anything changes.

**Comment by heitorlessa** [comment]
hey @Dilski thank you for flagging this inconsistency. I've pinged the X-Ray team [as there is no confirmed plan to deprecate X-Ray SDK](https://github.com/aws/aws-xray-sdk-node/issues/515#issuecomment-1172771486).

From our side, do you have ideas on where we can make this clearer? any specific place in the docs? FAQ?

Thanks a lot!! (@willfarrell too!)

**Comment by willfarrell** [comment]
Related to Tracing, there is a similar confusion with Metrics

- [aws-embedded-metrics](https://github.com/awslabs/aws-embedded-metrics-node)
- [AWS Powertools TypeScript](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/metrics/)
- [Otel](https://aws-otel.github.io/docs/getting-started/js-sdk/metric-manual-instr)

I was planning to deprecate using `aws-embedded-metrics` within middy (https://github.com/middyjs/middy/tree/main/packages/cloudwatch-metrics). Then a major release happened.

The dream from my perspective is that powertools is an easier to use wrapper around otel.

**Comment by heitorlessa** [comment]
Awesome, consider it done by EOW. If you don't mind, I'd appreciate a docs issue to help track this work: https://github.com/awslabs/aws-lambda-powertools-python/issues/new?assignees=&labels=documentation%2Ctriage&template=documentation_improvements.yml&title=Docs%3A+TITLE

Once we do in Python, we can easily transfer to other languages as the wording will be the same.

 > although it would be better if the X-Ray team were to update their warning/docs so we don't get the impression that we should just use adot/otel.

We'll pass that feedback along!

**Comment by niko-achilles** [comment]
Happy to read this issue today. Some comments from my side after 21 days.

> From there, if you want to use OTEL we should have an OTEL provider built-in, or if you want to use Provider X Y Z, you can bring their dependencies/config and plug that into our Tracer/Metrics/Logger pieces, keeping the same UX -- a thin-wrapper with our opinionated approach to handle 80% of use cases (20% being overrides, etc.).

Awesome @heitorlessa .
At the company i work for , the Teams in the modernization phase `Serverless first strategy` are now experimenting with Powertools (Python, Typescript, .NET) and in parallel with OTEL. 

The Developer experience of powertools (best practices built-in) is great. 
ADOT collector deployment strategy for serverless as the state of reality is today is not so developer friendly and has the domino effect in quality ;) Deploying a modified runtime as internal extension and and external extension for moving data from point A to point B gets complicated very fast.

The telemetry API announced recently is a better deployment approach in comparison to collector, however it is about moving telemetry data from Point A to Point B. ;)  

> (I'll write a RFC after re:Invent).

I am happy for the RFC moment after reinvent to read and also share experiences if needed.    

> We'll get there ;) I'm more hopeful than I was a year ago when looking at OTEL alone.

  I like this statement

**Comment by dreamorosi** [comment]
We are happy to announce that we are considering this feature and have published an RFC to ask your feedback: #90 

While we haven't committed to implement it just yet, we're actively seeking community feedback to shape OpenTelemetry support in Powertools for AWS Lambda.

We encourage all interested parties to provide feedback both in this RFC or via email at [aws-powertools-maintainers@amazon.com](mailto:aws-powertools-maintainers@amazon.com)

**Comment by npellegrin** [comment]
Regarding the [recent AWS anouncement about X-Ray SDKs/Daemon End-of-Support](https://aws.amazon.com/fr/blogs/mt/announcing-aws-x-ray-sdks-daemon-end-of-support-and-opentelemetry-migration/) will this feature soon become mandatory in order to continue having XRay traces with AWS Lambda Powertools ?
